File: pipeline/episode.py
"""Where each episode's files live (mirrors pipeline/episode.mjs)."""

import sys
from pathlib import Path

REPO = Path(__file__).resolve().parent.parent


def episode_arg(argv=sys.argv):
    return argv[argv.index("--episode") + 1] if "--episode" in argv else "renders"


def paths(ep=None):
    ep = ep or episode_arg()
    legacy = ep == "renders"
    return {
        "ep": ep,
        "script": REPO / ("script/narration.md" if legacy else f"script/{ep}.md"),
        "paragraphs": REPO / ("cache/paragraphs.json" if legacy else f"cache/{ep}/paragraphs.json"),
        "timeline": REPO / ("timing/timeline.json" if legacy else f"timing/{ep}/timeline.json"),
        "captions": REPO / ("timing/captions.vtt" if legacy else f"timing/{ep}/captions.vtt"),
        "narration": REPO / ("out/narration.wav" if legacy else f"out/{ep}/narration.wav"),
        "soundtrack": REPO / ("out/soundtrack.wav" if legacy else f"out/{ep}/soundtrack.wav"),
    }

File: pipeline/episode.mjs
// Where each episode's files live. Episode 1 ("renders") keeps its original
// paths; later episodes get their own directories.

import { join } from "node:path";

export function episodePaths(repo, ep = "renders") {
  const legacy = ep === "renders";
  return {
    ep,
    script: join(repo, legacy ? "script/narration.md" : `script/${ep}.md`),
    paragraphs: join(repo, legacy ? "cache/paragraphs.json" : `cache/${ep}/paragraphs.json`),
    timeline: join(repo, legacy ? "timing/timeline.json" : `timing/${ep}/timeline.json`),
    captions: join(repo, legacy ? "timing/captions.vtt" : `timing/${ep}/captions.vtt`),
    narration: join(repo, legacy ? "out/narration.wav" : `out/${ep}/narration.wav`),
    soundtrack: join(repo, legacy ? "out/soundtrack.wav" : `out/${ep}/soundtrack.wav`),
  };
}

// --episode <name> from argv (default "renders").
export function episodeArg(argv = process.argv) {
  const i = argv.indexOf("--episode");
  return i >= 0 ? argv[i + 1] : "renders";
}
